### DISCUSSION TITLE
Does the targets markdown work on RStudio Connect?

### DISCUSSION THREAD

**Discussion by 33Vito** [ropensci/targets#637] [Help] [answered]
Hi all, did anyone as experience using the target markdown on RStudio Connect? 

I am really a big fan of this package, and hope it would "just work" on the RStudio Connect which our IT department is working on. If it runs on the server, would the associated `_targets` folder be stored persistently or deleted once the document is generated?

Thanks

**Comment by wlandau** [comment, marked as answer]
You can, but it requires changing the location of the data store to something in `/mnt/shared-data/` (or `/car/`), e.g. `targets::tar_config_set(store = "/mnt/shared-data/_targets")`. cc @ian-flores, @jaredlander

**Reply by 33Vito** [reply]
Thanks Will. What if I just want to run the targets file via tar_make() on Studio Server？Do I have to save the _targets.R file in that shared-data folder as well？

**Reply by jaredlander** [reply]
The _targets.R file goes in your project directory. Changing where the cache is stored is only needed if you have Connect running the build or you just need storage somewhere else, perhaps for size reasons.

**Reply by 33Vito** [reply]
Hi, 

Just a follow-up. 

Since I am using the Target Markdown, I realised that I need to bundled the `_targets_r/` folder in the markdown yaml, then just use the normal targets store and scripts locations

```yaml
---
title: "Target Markdown"
output: html_document
resource_files:
- _targets_r/
---
```

```r
targets::tar_config_set(script = here::here("target/_targets.R"),
                        store = here::here("target/_targets"))
```

**Reply by 33Vito** [reply]
Under this approach, I don't have access to the cache data saved under `_targets` folder, but I would assume that RStudio Connect have a method for me to retrieve that `_targets` folder from the linux server? Otherwise my plan is to upload the `_targets` folder to a different place (e.g. sharepoint) at the end of my R Markdown to get a history of the cached files. It's not something I desperately want now, but might be useful in the future. 

Let me know what you thinks, @Will, @Jared

**Reply by jaredlander** [reply]
Bundling the cache in the deployed folder won't work with connect. This is because connect will not save the artifacts. So you need to save it somewhere external like somewhere in /mnt. And to make sure the connect machine account has access to it.

**Reply by 33Vito** [reply]
I see, thanks Jared. But I can save the cache to anywhere the server has access to, be it a network drive or sharepoint, I think that's enough for my purpose here. Cheers

**Comment by will** [comment]
> Let me know what you thinks, @will, @jared
> 


I’ve never used R really, so I’m probably not the best person to ask, but I can try to help. Maybe you could rsync the dir around from the server to where you need it instead of share point? Unless this is all windows stuff then I’m not sure if that’d make rsync more annoying. I’ve heard wsl2 is good though
